resources/Materials/TestSuite/stdlib/materials/material_node_discovery.mtlx

```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0"?>
+<materialx version="1.38">
+  <nodedef name="ND_material_def_material_1_0" node="material_def" version="1.0" isdefaultversion="true">
+    <output name="out" type="material" value="" />
+  </nodedef>
+  <standard_surface name="top_level_shader" type="surfaceshader" nodedef="ND_standard_surface_surfaceshader" version="1.0.1" xpos="620" ypos="950" />
+  <surfacematerial name="top_level_material_no_asssign" type="material" nodedef="ND_surfacematerial" xpos="925" ypos="926">
+    <input name="surfaceshader" type="surfaceshader" nodename="top_level_shader" />
+    <input name="displacementshader" type="displacementshader" value="" />
+  </surfacematerial>
+  <nodegraph name="top_level_material_in_graph_no_assign" xpos="925" ypos="1357" >
+    <standard_surface name="standard_surface2" type="surfaceshader" nodedef="ND_standard_surface_surfaceshader" version="1.0.1" xpos="-414.603" ypos="326.959" />
+    <surfacematerial name="surfacematerial1" type="material" nodedef="ND_surfacematerial" xpos="-67.8807" ypos="21.3596">
+      <input name="surfaceshader" type="surfaceshader" nodename="standard_surface2" />
+      <input name="displacementshader" type="displacementshader" value="" />
+    </surfacematerial>
+    <output name="out" type="material" nodename="surfacematerial1" />
+  </nodegraph>
+  <standard_surface name="top_level_shader1" type="surfaceshader" nodedef="ND_standard_surface_surfaceshader" version="1.0.1" xpos="10" ypos="34" />
+  <surfacematerial name="top_level_material_assigned" type="material" nodedef="ND_surfacematerial" xpos="315" ypos="10">
+    <input name="surfaceshader" type="surfaceshader" nodename="top_level_shader1" />
+    <input name="displacementshader" type="displacementshader" value="" />
+  </surfacematerial>
+  <nodegraph name="NG_material_def_material_1_0" xpos="925" ypos="1596" nodedef="ND_material_def_material_1_0">
+    <standard_surface name="standard_surface2" type="surfaceshader" nodedef="ND_standard_surface_surfaceshader" version="1.0.1" xpos="-414.603" ypos="326.959" />
+    <surfacematerial name="surfacematerial1" type="material" nodedef="ND_surfacematerial" xpos="-67.8807" ypos="21.3596">
+      <input name="surfaceshader" type="surfaceshader" nodename="standard_surface2" />
+      <input name="displacementshader" type="displacementshader" value="" />
+    </surfacematerial>
+    <output name="out" type="material" nodename="surfacematerial1" />
+  </nodegraph>
+  <material_def name="top_level_material_def" type="material" nodedef="ND_material_def_material_1_0" version="1.0" xpos="925" ypos="1787" />
+  <nodegraph name="top_level_material_in_graph_assigned" xpos="315" ypos="468" ldx_io_nodes="&amp;lt;?xml version=&quot;1.0&quot;?&amp;gt;&#10;&amp;lt;materialx version=&quot;1.38&quot;&amp;gt;&#10;  &amp;lt;layout name=&quot;io&quot; type=&quot;&quot; /&amp;gt;&#10;&amp;lt;/materialx&amp;gt;&#10;">
```

Comment:
What is the meaning of the `ldx_io_nodes` attribute?  Perhaps it's an application specific artifact that we should remove from the test suite?

Reply:
Application artifact. Removed.